### PR TITLE
Add checksum for py-hepdata-lib 0.10.1, py-hepdata-validator 0.3.3

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-hepdata-lib/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyHepdataLib(PythonPackage):
     homepage = "https://github.com/HEPData/hepdata_lib"
     pypi = "hepdata_lib/hepdata_lib-0.9.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-hepdata-lib/package.py]: Does this even build from the source code from PyPI? It looks like the tarball is missing the `requirements.txt` file.

File: var/spack/repos/builtin/packages/py-hepdata-lib/package.py [hunk 2]
```diff
@@ -22,3 +23,4 @@ class PyHepdataLib(PythonPackage):
     depends_on("py-six", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-hepdata-lib/package.py:25]: I think pyyaml may actually be `@5:`

File: var/spack/repos/builtin/packages/py-hepdata-validator/package.py [hunk 3]
```diff
@@ -22,6 +23,9 @@ class PyHepdataValidator(PythonPackage):
     version("0.1.8", sha256="08686563e0130c5dd6d9fb8d5c7bf5a2617a637b105a42f7106b96a31eaffa61")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-hepdata-validator/package.py:24]: Python 3.6+ required now

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-hepdata-lib/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-hepdata-lib/package.py]: Yep, it does not. Fixed.

File: var/spack/repos/builtin/packages/py-hepdata-lib/package.py [hunk 4]
```diff
@@ -10,15 +10,20 @@ class PyHepdataLib(PythonPackage):
...
 
-    version("0.9.0", sha256="b7b194b8af0428f34094ac403f8794a672c82d85e33154161d6b3744cc2b9896")
+    version("0.10.1", sha256="71c635963883c51e7be18e03d80bfe42c5de350852b01010e3e45cbd1bff7a81")
+    version("0.9.0", sha256="c9238e45c603d7c061ed670cf197ff03ad9d370ab50419b6916fda2cd86d6150")
 
     depends_on("py-setuptools", type="build")
     depends_on("root+python", type=("build", "run"))
     depends_on("py-numpy", type=("build", "run"))
     depends_on("py-pyyaml@4:", type=("build", "run"))
+    depends_on("py-pyyaml@5:", when="@0.10.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-hepdata-lib/package.py]: I think the old version also required 5+

**Review by adamjstewart — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Pull request merged**